homeassistant/components/person/__init__.py

```diff
@@ -373,18 +377,34 @@ def _async_handle_tracker_update(self, entity, old_state, new_state):
         """Handle the device tracker state changes."""
         self._update_state()
 
+    def _get_latest(self, prev, curr):
+        return curr \
+               if prev is None or curr.last_updated > prev.last_updated \
+               else prev
+
     @callback
     def _update_state(self):
         """Update the state."""
-        latest = None
+        latest_home = latest_not_home = latest_gps = latest = None
         for entity_id in self._config.get(CONF_DEVICE_TRACKERS, []):
             state = self.hass.states.get(entity_id)
 
             if not state or state.state in IGNORE_STATES:
                 continue
 
-            if latest is None or state.last_updated > latest.last_updated:
-                latest = state
+            if state.attributes.get(ATTR_SOURCE_TYPE) == SOURCE_TYPE_GPS:
```

Comment:
Checking source GPS first, before state home, doesn't match the PR description. Is the logic correct?

Reply:
@MartinHjelmare, yes. If there are other non-GPS devices, which shown 'home', they'll be got prior to GPS in next conditions block.

Reply:
I think this should be clarified in the PR description and in the docs, if it's not done already.

Also, it would help to add examples that explain why this logic is better then the previous logic.

Reply:
Done in home-assistant/home-assistant.io#8779